Index: app/controllers/exams_controller.rb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/controllers/exams_controller.rb	(date 1405087689000)
+++ app/controllers/exams_controller.rb	(revision )
@@ -96,11 +96,14 @@
     (archive == 'JHHPeer' || archive == 'JHHP') ? 'JHH' : archive
   end
 
+  def value_present(val)
+    val and val.present?
+  end
   # for ajax: 200 or 300 is hardcoded. Be careful.
   def resolve_context
     #returns list of java objects
 
-    if  (!params[:patient_acc] or params[:patient_acc].blank?) and (!params[:patient_mrn] or params[:patient_mrn].blank?)
+    if  !value_present(params[:patient_acc]) and !value_present(params[:patient_mrn])
       render :json => {}, status: 200
     end
 
@@ -110,9 +113,9 @@
       # exams = exams.map(&:harbinger_exam)
       audit(:rad_exams, exams)
       @exams = exams
-      if exams.size > 1
+      if (exams.size > 1) or  (!value_present(params[:patient_acc]) or !value_present(params[:patient_mrn]))
         render :layout => false, status: 300
-      elsif exams.size ==1
+      elsif exams.size == 1
         aiExam = AiExam.new(:harbinger_exam => exams.first, :em => get_entity_manager)
         render :json => Java::HarbingerSdk::DataUtils.to_json(aiExam.pretty), status: 200
       end
Index: .rvmrc
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .rvmrc	(date 1405087689000)
+++ .rvmrc	(revision )
@@ -1,3 +1,3 @@
 rvm use jruby-1.7.12@peer_review --create
 
-export JRUBY_OPTS="-J-XX:MaxPermSize=256m -J-Dharbinger.sso.url='http://ai.rad.jhmi.edu:8088/openam' -J-Dharbinger.sso.login=https://ai.rad.jhmi.edu:8443/openam/spssoinit?metaAlias=/spmeta&idpEntityID=https://bluejay2.johnshopkins.edu/idp/shibboleth&NameIDFormat=transient&RelayState= -J-Dharbinger.sso.logout=https://ai.rad.jhmi.edu:8443/openam/logout --1.9 "
+export JRUBY_OPTS="-J-XX:MaxPermSize=256m -J-Dharbinger.sso.url='http://ai.rad.jhmi.edu:8088/openam' -J-Dharbinger.sso.login=https://ai.rad.jhmi.edu:8443/openam/spssoinit?metaAlias=/spmeta&idpEntityID=https://bluejay2.johnshopkins.edu/idp/shibboleth&NameIDFormat=transient&RelayState= -J-Dharbinger.sso.logout=https://ai.rad.jhmi.edu:8443/openam/logout --1.9"
